### ISSUE TITLE
Template params break url encoded strings

### ISSUE THREAD

**Issue by pimlie** [unjs/unhead#127]
### Environment

main branch

### Reproduction

```patch
diff --git a/test/fixtures.ts b/test/fixtures.ts
index 111dc51..56bf241 100644
--- a/test/fixtures.ts
+++ b/test/fixtures.ts
@@ -38,6 +38,10 @@ export const basicSchema: Head = {
     {
       charset: 'utf-8',
     },
+    {
+      property: 'twitter:image',
+      content: 'https://cdn.example.com/some%20image.jpg'
+    }
   ],
   link: [
     {
```

### Describe the bug

The templateParams plugin incorrectly assumes the `%20` is a template parameter and tries to replace it, causing the content value to be truncated to just `https://cdn.example.com/some`

![image](https://user-images.githubusercontent.com/1067403/224700943-b69dc943-977e-47c1-98f0-21ea9bb30f1e.png)

### Additional context

Possible solutions:

- decode strings before searching for template params
- ignore template params for which no replacement value was configured 

### Logs

```sh
FAIL  test/unhead/ssr/ssr.test.ts > ssr > basic
AssertionError: Snapshot `ssr > basic 1` mismatched
 ❯ test/unhead/ssr/ssr.test.ts:18:17
     16| 
     17|     const ctx = await renderSSRHead(head)
     18|     expect(ctx).toMatchInlineSnapshot(`
       |                 ^
     19|       {
     20|         "bodyAttrs": " class=\\"dark\\"",

  - Expected  - 0
  + Received  + 1

      bodyTags": "",
      "bodyTagsOpen": "",
      "headTags": "<meta charset="utf-8">
    <script src="https://cdn.example.com/script.js"></script>
  + <meta property="twitter:image" content="https://cdn.example.com/some">
    <link rel="icon" type="image/x-icon" href="https://cdn.example.com/favicon.ico">",
      "htmlAttrs": " lang="de"",
    }
```

**Comment by harlan-zw** [comment]
Thanks for the detailed issue @pimlie 

This is fixed in 1.1.23

**Comment by pimlie** [comment]
Awesome, can confirm it works. Thx for the quick fix!
